cmd/minishift/cmd/start.go

```diff
@@ -108,7 +109,12 @@ For the latter see 'minishift config -h'.`,
 		Run: runStart,
 	}
 
+	// enable experimental flags
+	//startCmd.Flags().BoolVar(&future, "future", false, "Enable experimental features (unsupported)")
```

Comment:
so why not have a '--show-experimental-flags' as a global flag (not just in the start command). If the flag is set we show them, otherwise not. In this case we don't need environment variables. 

We keep a set of experimental flags in shared_flags and in root.go, we check the state of --show-experimental-flags. If true, we how all otherwise hidden flags.